syntax = "proto3";
package pb;

message LogContent
{
  string key = 1;
  string value = 2;
}

message Log
{
  int64 time = 1;  // UNIX Time Format
  repeated LogContent contents = 2;
  oneof OptionalTimeNs {
    fixed32 TimeNs = 3;  // 纳秒级时间戳
  }
}

message LogTag
{
  string key = 1;
  string value = 2;
}

message LogGroup
{
  repeated Log logs = 1;
  string source = 2;
  repeated LogTag log_tags = 3;
  string filename = 4;
  string context_flow = 5;  // 该字段暂无效用
}

message LogGroupList
{
  repeated LogGroup log_groups = 1;
}
